internal/controller/spannerautoscaler_controller_test.go

```diff
@@ -410,3 +425,57 @@ var _ = Describe("Get and overwrite scaledown interval", func() {
 		Expect(got).To(Equal(want))
 	})
 })
+
+var _ = Describe("Get and overwrite scaleup interval", func() {
+	var testReconciler *SpannerAutoscalerReconciler
+	controllerScaleUpInterval := 55 * time.Minute
+
+	BeforeEach(func() {
+		By("Creating a test reconciler")
+		testReconciler = &SpannerAutoscalerReconciler{
+			scaleUpInterval: controllerScaleUpInterval,
+			clock:           testingclock.NewFakeClock(fakeTime),
+			log:             logr.Discard(),
+		}
+	})
+
+	It("should get controller default scaleup interval", func() {
+		want := controllerScaleUpInterval
+		sa := &spannerv1beta1.SpannerAutoscaler{
+			Status: spannerv1beta1.SpannerAutoscalerStatus{
+				LastScaleTime:          metav1.Time{Time: fakeTime.Add(-time.Minute)},
+				CurrentProcessingUnits: 1000,
+				DesiredProcessingUnits: 2000,
+				InstanceState:          spannerv1beta1.InstanceStateReady,
+			},
+			Spec: spannerv1beta1.SpannerAutoscalerSpec{
+				ScaleConfig: spannerv1beta1.ScaleConfig{},
+			},
+		}
+		got := getOrConvertTimeDuration(sa.Spec.ScaleConfig.ScaleupInterval, testReconciler.scaleUpInterval)
+		Expect(got).To(Equal(want))
+	})
+
+	It("should override default scaleup interval with custom SpannerAutoscaler configuration value", func() {
+		want := 20 * time.Minute
+		scaleupInterval := metav1.Duration{
+			Duration: want,
+		}
+		sa := &spannerv1beta1.SpannerAutoscaler{
+			Status: spannerv1beta1.SpannerAutoscalerStatus{
+				LastScaleTime:          metav1.Time{Time: fakeTime.Add(-time.Minute)},
+				CurrentProcessingUnits: 2000,
+				DesiredProcessingUnits: 1000,
+				InstanceState:          spannerv1beta1.InstanceStateReady,
+			},
+			Spec: spannerv1beta1.SpannerAutoscalerSpec{
+				ScaleConfig: spannerv1beta1.ScaleConfig{
+					ScaleupInterval: &scaleupInterval,
+				},
+			},
+		}
+
+		got := getOrConvertTimeDuration(sa.Spec.ScaleConfig.ScaleupInterval, testReconciler.scaleUpInterval)
+		Expect(got).To(Equal(want))
+	})
+})
```

Comment:
This test case just seems to be checking the `getOrConvertTieDuration` function response. If so, maybe we can have a simpler test case just for that function, which doesn't need any reconciler or SpannerAutoscaler resources.

Reply:
I made it simpler in https://github.com/mercari/spanner-autoscaler/pull/132/commits/b45344889c6e8c84c8f827021bb9169517d945e8